### PR TITLE
[Docs] - Added Docs for mesheryctl Commands and Updates Title for Doc Pages

### LINKED ISSUE #2941: [Docs] `mesheryctl system context delete` page missing
#### Current State
No page exists for this command in the [mesheryctl Command Reference](https://docs.meshery.io/reference/mesheryctl).

#### Desired State
A docs page for this subcommand should be available.

#### Implementation
1. Create an entry for this subcommand in https://github.com/layer5io/meshery/tree/master/docs/_data/mesheryctlcommands/

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### LINKED ISSUE #2942: [Docs] `mesheryctl system context switch` page missing
#### Current State
No page exists for this command in the [mesheryctl Command Reference](https://docs.meshery.io/reference/mesheryctl).

#### Desired State
A docs page for this subcommand should be available.

#### Implementation
1. Create an entry for this subcommand in https://github.com/layer5io/meshery/tree/master/docs/_data/mesheryctlcommands/

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### LINKED ISSUE #2944: [Docs] `mesheryctl system completion` page missing
#### Current State
No page exists for this command in the [mesheryctl Command Reference](https://docs.meshery.io/reference/mesheryctl).

#### Desired State
A docs page for this subcommand should be available.

#### Implementation
1. Create an entry for this subcommand in https://github.com/layer5io/meshery/tree/master/docs/_data/mesheryctlcommands/

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### LINKED ISSUE #2946: [Docs] `mesheryctl pattern apply` page missing
#### Current State
No page exists for this command in the [mesheryctl Command Reference](https://docs.meshery.io/reference/mesheryctl).

#### Desired State
A docs page for this subcommand should be available.

#### Implementation
1. Create an entry for this subcommand in https://github.com/layer5io/meshery/tree/master/docs/_data/mesheryctlcommands/

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### LINKED ISSUE #3057: [Docs] - Change the title for `mesheryctl` command reference pages
#### Description
The mesheryctl command reference pages needs better titles for the respective commands

#### Expected Behavior

All the pages have improved titles according to the below convention,

- `| Meshery` to be appended for every command
- `mesheryctl system context | Meshery` or `context - mesheryctl | Meshery`

#### Screenshots/Logs

Before:

image

After:
image

#### To Reproduce
1. Go to 'docs.meshery.io'
2. Click on 'mesheryctl command reference'
3. Go to any page under command reference Eg: https://docs.meshery.io/reference/mesheryctl/mesheryctl/version
4. See title bar for the issue

### PR DESCRIPTION
**Pull request by adithyaakrishna**
Signed-off-by: Adithya Krishna <aadithya794@gmail.com>

**Description**

This PR updates `mesheryctl` docs

- Fixes #2946
- Fixes #2944 
- Fixes #2941
- Fixes #2942
- Fixes #3057 

This PR also updates mesheryctl pattern commands to `mesheryctl pattern` from `mesheryctl exp pattern`

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by adithyaakrishna** [comment]
Also this is the 3000th PR 🎉

**Comment by alphaX86** [comment]
I love you 3000!! ❤️😂

**Comment by alphaX86** [comment]
@adithyaakrishna is this intended?
image

I guess you didn't consider rowspan here....

**Comment by alphaX86** [comment]
Same way here...
image

**Comment by alphaX86** [comment]
Uhh... you mentioned some issues will fix right? But I'm afraid you didn't do so....

#2944 
image

#2946 
image

`switch` and `delete` are fine but some formatting is needed in table

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-apply.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+layout: default
+title: apply
+permalink: reference/mesheryctl/pattern/apply
+type: reference
```
Comment (alphaX86) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-apply.md:5]: You've created and it's great but it must be reflected on the table also...

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-sys-completion.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+---
+layout: default
+title: completion
+permalink: reference/mesheryctl/system/completion
+type: reference
```
Comment (alphaX86) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-sys-completion.md:5]: Same way here...

File: docs/pages/reference/mesheryctl-commands.md [hunk 3]
```diff
@@ -180,31 +180,31 @@ Installation, troubleshooting and debugging of Meshery and its adapters.
           <td>{{ flag.description }}</td>
         </tr>
       {% endfor %}
-      {% for subcommand_hash in site.data.mesheryctlcommands.lifecycle.system-context.subcommands %}{% assign subcommand = subcommand_hash[1] %}
+      {% for subcommand_hash in site.data.mesheryctlcommands.lifecycle.system-context.view.command %}{% assign subcommand = subcommand_hash[1] %}
         <tr>
-          <td>{{ subcommand.name }}</td>
+          <td rowspan=3><a href="{{ site.baseurl }}/reference/mesheryctl/system/context/view">{{ subcommand.name }}</a></td>
```
Comment (alphaX86) [inline docs/pages/reference/mesheryctl-commands.md:185]: Just look at rowspan at line no 69 (this is just to show...)

**Review by alphaX86 — CHANGES_REQUESTED** [review]
Make the above changes I suggested :)

**Comment by adithyaakrishna** [comment]
@alphaX86 There are some commits locally I didn't push yet, they will fix those issues 😶

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-apply.md (hunk 1 above)
Comment (adithyaakrishna) [reply docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-apply.md:5]: Yes, it will be. Have created a local comkit didn't push them yet

**Comment by alphaX86** [comment]
> @alphaX86 There are some commits locally I didn't push yet, they will fix those issues 😶

Ok no issues xD Just said my opinion ;)

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-sys-completion.md (hunk 2 above)
Comment (adithyaakrishna) [reply docs/pages/reference/mesheryctl/subcommands/mesheryctl-sys-completion.md:5]: This one too, will push the commit today

**Review by navendu-pottekkat — CHANGES_REQUESTED** [review]
@adithyaakrishna Some issues-

Completion showing an 404-
 
Screenshot from 2021-05-24 00-12-50

Two instances of completion-

Screenshot from 2021-05-24 00-13-09

Could you add more to the pattern apply page? Please refer to the [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0) spreadsheet for it.

Let me know if you still have doubts on what to add. FYI @piyushsingariya

**Review by alphaX86 — CHANGES_REQUESTED** [review]
@adithyaakrishna seems you didn't consider what change I meant...
image

Look the above one, the rowspan is off here

**Comment by adithyaakrishna** [comment]
@alphaX86 Ohh this is the screenshot from docs.meshery.io

image

Will fix that too, The current one has status and update aligned to the left which needs to be put to the right. My version brings log and restart to the left too 😶

**Comment by alphaX86** [comment]
> @alphaX86 Ohh this is the screenshot from docs.meshery.io
> 
> Will fix that too, The current one has status and update aligned to the left which needs to be put to the right. My version brings log and restart to the left too 😶

Huh??? So, it means that I failed to check in previous PR #2974 IG. It's my fault.... Anyway, it's great that you've said this first. Sorry dude 😓

**Comment by adithyaakrishna** [comment]
> > @alphaX86 Ohh this is the screenshot from docs.meshery.io
> > Will fix that too, The current one has status and update aligned to the left which needs to be put to the right. My version brings log and restart to the left too 😶
> 
> Huh??? So, it means that I failed to check in previous PR #2974 IG. It's my fault.... Anyway, it's great that you've said this first. Sorry dude 😓

No no, I'm not saying its your fault. The whole thing is very confusing. `rowspan` isnt the problem here. The thing is some files use `subcommand_hash` and some `command_hash`, thats the first point. Due to this the for loop considers `command_hash` as the main command instead of that being the subcommand. Due to this the lines are getting messed up a bit

**Comment by alphaX86** [comment]
Ah I get it! I'm considering to revamp the whole YAML structure.. NOT NOW! 🤣 I have upcoming exams and will be busy for a week and later for June 😭

**Comment by leecalcote** [comment]
@adithyaakrishna please signal when you're ready for another round of review.

**Comment by adithyaakrishna** [comment]
> @adithyaakrishna please signal when you're ready for another round of review.

Yess, I have made some commits locally and still teeny tiny changes are remaining. Will push them today

**Comment by adithyaakrishna** [comment]
@leecalcote Ready To Merge 🎉

**Comment by alphaX86** [comment]
image

This link throws 404 error actually

**Comment by alphaX86** [comment]
One flag is actually added to this, we can test it during tomorrow and see about it
image

**Comment by alphaX86** [comment]
As usual.... :)
image

**Comment by alphaX86** [comment]
Anyways, it's a great improvement from previous one @adithyaakrishna Good job 👍

**Comment by adithyaakrishna** [comment]
Ohh The issue with fixing the proper margins 🤦‍♂️
That's a bit strange, it was running locally, but will give this a look now

File: docs/pages/reference/mesheryctl-commands.md [hunk 4]
```diff
@@ -79,11 +79,11 @@ Installation, troubleshooting and debugging of Meshery and its adapters.
       </tr>
     {% endfor %}
     {% for command_hash in site.data.mesheryctlcommands.lifecycle.system.start.command %}{% assign command = command_hash[1] %}
-    <tr>
-      <td rowspan=4><a href="{{ site.baseurl }}/reference/mesheryctl/system/start">{{ command.name }}</a></td>
-      <td></td>
-      <td>{{ command.description }}</td>
-    </tr>
+      <tr>
+        <td rowspan=4><a href="{{ site.baseurl }}/reference/mesheryctl/system/start">{{ command.name }}</a></td>
```
Comment (alphaX86) [inline docs/pages/reference/mesheryctl-commands.md:83]: Refer Line 69 please!! (this is just for example I'm doing at line 83)
```suggestion
        <td rowspan=16><a href="{{ site.baseurl }}/reference/mesheryctl/system">{{ command.name }}</a></td>
```

**Review by alphaX86 — CHANGES_REQUESTED** [review]

**Comment by adithyaakrishna** [comment]
@alphaX86 The margins issue, might need another fix even after this PR gets merged as a new page is added by @Anita-ihuman 's PR i.e. PR #3022 
I'll look into the 404 Thing wrt completion page

**Comment by adithyaakrishna** [comment]
@alphaX86 Done with the changes 🚀

**Review by navendu-pottekkat — COMMENTED** [review]
1. The system completion example could be made more specific.

Generally we use it like `source <(mesheryctl system completion zsh)` and not just run `system completion`.

You could run `mesheryctl system completion --help` to get some idea.

2. `pattern apply` page is not shown on the table-

Screenshot from 2021-05-28 10-07-06

**Comment by adithyaakrishna** [comment]
The above two commits `4f9cd4e` & `af8c0b2` updates the title for fixing #3057

**Review by navendu-pottekkat — APPROVED** [review]
Looks great! Thank you @adithyaakrishna 💯

**Comment by leecalcote** [comment]
We desperately need to get away from repeating ourselves in the mesheryctl docs. Transitioning to a consolidated yaml datafile like https://github.com/layer5io/meshery/blob/leecalcote/docs/mesheryctl/docs/_data/mesheryctlcommands/cmds.yml would do the trick.

**Comment by adithyaakrishna** [comment]
Yayyyy, PR 3000 is Merged finally 💥🥳

Thanks @navendu-pottekkat & @alphaX86 🍻

**Comment by alphaX86** [comment]
> We desperately need to get away from repeating ourselves in the mesheryctl docs. Transitioning to a consolidated yaml datafile like https://github.com/layer5io/meshery/blob/leecalcote/docs/mesheryctl/docs/_data/mesheryctlcommands/cmds.yml would do the trick.

We need to work this.... After sorting out some work, I or some one will probably take care of this

**Pull request merged**